File: SchedulingFront-OLD/src/app/models/product.ts
import { ProductType } from './productType';
import { BasePaging } from '../common/models/basePaging';
import { SelectListModel } from '../common/models/selectListModel';

export class Product {
    productId: number;
    code: string;
    name: string;
    image: string;
    active: boolean;
    productTypeid: number;
    productType: ProductType;
    price: number;
    organizationUnitId: number;
    organizationUnits = new Array<number>();
    employees: Array<number>;
    productPricelist = new Array<ProductPricelist>();
    organizationUnitsString: string;
}

export class ProductPricelist {
    organizationUnitId: number;
    organizationUnitName: string;
    documentDetailId: number;
    price: number;
    documentId: number;
    productId: number;
    name: string;
    employeeId: number;
    constructor(organizationUnitId: number = null, organizationUnitName: string = null) {
        this.organizationUnitId = organizationUnitId;
        this.organizationUnitName = organizationUnitName;
        this.documentDetailId = null;
        this.price = null;
        this.documentId = null;
    }
}

export class ProductPaging extends BasePaging {
    name: string;
    code: string;
    productTypeId: number;
    organizationUnits = new Array<number>();
}

export class ProductSelectList {
    products: Array<Product>;
    productPricelist = new Array<ProductPricelist>();

}
export class ProductSelectListInput {
    allOrgUnits: boolean;
    organizationUnits = new Array<number>();
    constructor(organizationUnits: Array<number>, allOrgUnits = false) {
        this.organizationUnits = organizationUnits;
        this.allOrgUnits = allOrgUnits;
    }
}

export class ProductSelectListModel extends SelectListModel {
    price: number;
    employeeId: number;
    constructor(id: number, name: string, price: number, employeeId: number) {
        super(id, name);
        this.price = price;
        this.employeeId = employeeId;
    }
}

File: SchedulingAPI/Scheduling.Contract/Models/Product.cs
using Entity.Base;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;

namespace Scheduling.Contract.Models
{
    public class Product
    {
        [PrimaryKey]
        public long? ProductId { get; set; }
        public int ProductTypeId { get; set; }
        public ProductType ProductType { get; set; }
        public string Code { get; set; }
        public string Name { get; set; }
        public string Image { get; set; }
        public bool Active { get; set; }

    }
    public class ProductPaging : BasePaging
    {
        public string Name { get; set; }
        public string Code { get; set; }
        public List<int> OrganizationUnits = new List<int>();
        public List<int> ProductTypes = new List<int>();
    }

    public class ProductSelectList
    {
        public Product Product { get; set; }
        public decimal Price { get; set; }
        public int OrganizationUnitId { get; set; }
        public int EmployeeId { get; set; }
    }

    //public class ProductSelectList
    //{
    //    public IEnumerable<Product> Products { get; set; }
    //    public IEnumerable<ProductPricelist> ProductPricelist { get; set; }
    //    public ProductSelectList(IEnumerable<Product> products, IEnumerable<ProductPricelist> productPricelist)
    //    {
    //        Products = products;
    //        ProductPricelist = productPricelist;
    //    }
    //}
    //public class ProductPricelist
    //{
    //    public long? ProductId { get; set; }
    //    public string Name { get; set; }
    //    public long? DocumentId { get; set; }
    //    public int? OrganizationUnitId { get; set; }
    //    public string OrganizationUnitName { get; set; }
    //    public decimal? Price { get; set; }
    //    public long? DocumentDetailId { get; set; }
    //    public int EmployeeId { get; set; }
    //}

    //public class ProductSelectList
    //{
    //    public IEnumerable<Product> Products { get; set; }
    //    public IEnumerable<ProductPricelist> ProductPricelist { get; set; }
    //    public ProductSelectList(IEnumerable<Product> products, IEnumerable<ProductPricelist> productPricelist)
    //    {
    //        Products = products;
    //        ProductPricelist = productPricelist;
    //    }
    //}
}